### ISSUE TITLE
bug(material/dialog): auto-focusing a CSS class doesn't work when animations are disabled

### ISSUE THREAD

**Issue by mmalerba** [angular/components#29194]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Auto-focusing a CSS class fails when animations are disabled

### Reproduction

StackBlitz link: https://stackblitz.com/edit/yuzht5-trwt8i?file=src%2Fmain.ts
Steps to reproduce:
1. Open dialog
2. Focus is not on install button

### Expected Behavior

Auto-focusing a CSS class works regardless of animations

### Actual Behavior

Auto-focusing a CSS class fails when animations are disabled

### Environment

- Angular: 18.0.0
- CDK/Material: 18.0.0
